float64[] center
float64[] length
float64 area